### ISSUE TITLE
PulseBlock ⊗ Pulse recursion leads to stack overflow

### ISSUE THREAD

**Issue by caryan** [BBN-Q/QGL.jl#24]
Something in #23 broke `PulseBlock` ⊗ `Pulse`

```julia
julia> using QGL
julia> q1 = Qubit("q1")
julia> q2 = Qubit("q2")
julia> QGL.PulseBlock(X(q1)) ⊗ Y(q2)
ERROR: StackOverflowError:
 in ⊗(::QGL.PulseBlock, ::QGL.Pulse) at /home/cryan/.julia/v0.5/QGL/src/pulses.jl:161 (repeats 80000 times)
```

**Comment by caryan** [comment]
Seems `promote_rule` doesn't pick up `Abstract` types

**Comment by caryan** [comment]
Closed by 28a65b0525435f3ab2bd0f4a341fe989e017b29
